var/spack/repos/builtin/packages/openfoam/package.py

```diff
@@ -918,7 +918,12 @@ class OpenfoamArch:
 
     #: Map spack compiler names to OpenFOAM compiler names
     #  By default, simply capitalize the first letter
-    compiler_mapping = {"aocc": "Amd", "fj": "Fujitsu", "intel": "Icc", "oneapi": "Icx"}
+    compiler_mapping = {
+        "aocc": "Amd",
+        "fj": "Fujitsu",
+        "intel": "Icc",
+        "intel-oneapi-compilers": "Icx",
```

Comment:
Ok here is a potentially nasty issue -- some compilers can be provided different ways. In particular, `intel-oneapi-compilers` is provided by `intel-oneapi-compilers` *and* by intel-parallel-studio. So I think this is incomplete. How do we allow a user to refer to `oneapi` in any of its forms?

Reply:
Isn't `intel-parallel-studio` deprecated? Why do you say that `intel-parallel-studio` provides oneapi compilers?

Reply:
See https://github.com/spack/spack/pull/45188 which will remove Intel packages that have been deprecated in 2023. 

In any case we don't have issues if two packages provide the same compiler, in the sense that it will either be:
```
pkg_a %provider1
```
vs.
```
pkg_b %provider2
```
for legacy compilers `%intel` is provided by `intel-oneapi-compilers-classic`.